src/hooks/useEventBusForStatistic.ts

```diff
@@ -1,16 +1,140 @@
-import { EVENT_NAMES_FROM_STATISTIC } from "../events";
+import {
+  ActionToStatistic,
+  EVENT_NAMES_FROM_STATISTIC,
+  EVENT_NAMES_TO_STATISTIC,
+  isReceiveDataForStatistic,
+  isReceiveDataForStatisticError,
+} from "../events";
 import { usePostMessage } from "./usePostMessage";
+import { useCallback, useEffect, useReducer } from "react";
+import { StatisticData } from "../services/refact";
+
+type StatisticState = {
+  statisticData: StatisticData | null;
+  isLoading: boolean;
+  error: string;
+};
+
+function createInitialState(): StatisticState {
+  return {
+    statisticData: null,
+    isLoading: true,
+    error: "",
+  };
+}
+
+const initialState = createInitialState();
+
+function reducer(
+  state: StatisticState,
+  action: ActionToStatistic,
+): StatisticState {
+  switch (action.type) {
+    case EVENT_NAMES_TO_STATISTIC.REQUEST_STATISTIC_DATA:
+      return {
+        ...state,
+        isLoading: true,
+        error: "",
+      };
+    case EVENT_NAMES_TO_STATISTIC.RECEIVE_STATISTIC_DATA:
+      return {
+        ...state,
+        statisticData: action.payload
+          ? (action.payload as StatisticData)
```

Comment:
if you use an `if`s  instead of a switch you can use type guards instead of casting :)

Reply:
done